geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/metadata/JdbcMetadata.scala

```diff
@@ -65,21 +67,34 @@ import org.opengis.feature.simple.SimpleFeatureType
   * @param pool connection pool
   * @param root storage root path
   * @param sft simple feature type
-  * @param encoding encoding
-  * @param scheme partition scheme
-  * @param leafStorage leaf storage
+  * @param meta basic metadata config
   **/
 class JdbcMetadata(
     pool: PoolingDataSource[PoolableConnection],
     root: String,
     val sft: SimpleFeatureType,
-    val encoding: String,
-    val scheme: PartitionScheme,
-    val leafStorage: Boolean
+    meta: Metadata
   ) extends StorageMetadata {
 
   import JdbcMetadata.PartitionsTable
 
+  import scala.collection.JavaConverters._
+
+  override val scheme: PartitionScheme = PartitionSchemeFactory.load(sft, meta.scheme)
+  override val encoding: String = meta.config(Metadata.Encoding)
+  override val leafStorage: Boolean = meta.config(Metadata.LeafStorage).toBoolean
+
+  private val kvs = new ConcurrentHashMap[String, String](meta.config.asJava)
+
+  override def get(key: String): Option[String] = Option(kvs.get(key))
+
+  override def set(key: String, value: String): Unit = {
```

Comment:
Sanity check question:  Do we have a unit tests which cover changes to the FSDS JDBC metdata, etc, etc?

Reply:
we do actually have a unit test, i will add some new test cases for these changes.

Reply:
and... the test failed so good catch :)

Reply:
Whew!  Touchdown reviewer!  Give me enough time and even I'll get lucky and turn over a bug!